app/models/concerns/orders_package_actions.rb

```diff
@@ -0,0 +1,120 @@
+module OrdersPackageActions
+  extend ActiveSupport::Concern
+
+
+  #
+  # Enum of all available actions
+  #
+  module Actions
+    class Action < Utils::Toggleable
+      def initialize(name, &block)
+        super(name)
+        @block = block
+      end
+
+      def run(orders_package, opts = {})
+        @block.call(orders_package, opts)
+      end
+    end
+
+    CANCEL          = Action.new('cancel') { |op| op.cancel }
+    DISPATCH        = Action.new('dispatch') { |op| op.dispatch_orders_package }
+    UNDISPATCH      = Action.new('undispatch') { |op| op.undispatch_orders_package }
+    REDESIGNATE     = Action.new('redesignate') { |op, opts| op.redesignate(opts[:order_id]) }
+    EDIT_QUANTITY   = Action.new('edit_quantity') { |op, opts| op.edit_quantity(opts[:quantity]) }
+
+    ALL_ACTIONS = [CANCEL, DISPATCH, UNDISPATCH, REDESIGNATE, EDIT_QUANTITY]
+  end
+
+  #
+  # Returns a list of actions which can be run against the orders_package
+  #
+  def allowed_actions
+    ActionResolver.resolve(self)
+  end
+
+  #
+  # Returns true if it is possible to run the action
+  #
+  def can_exec_action(name)
+    allowed_actions
+      .select { |act| act[:enabled] }
+      .map { |act| act[:name] }
+      .include? name.to_s
+  end
+
+  #
+  # Tries to run the action
+  #
+  def exec_action(name, opts = {})
+    unless can_exec_action(name)
+      raise ArgumentError.new(I18n.t('orders_package.action_disabled', name: name))
+    end
+
+    action = Actions::ALL_ACTIONS.find { |act| act.name.to_s == name.to_s }
+    action.run(self, opts)
+  end
+
+  private
```

Comment:
Is this private keyword is required here?

Reply:
It separates the internal helper methods from the exposed API